full-stack-tests/presentation/src/learning-snippets/Content.test.ts

```diff
@@ -11,582 +11,1002 @@ import { getFieldByLabel, tryGetFieldByLabel } from "../Utils";
-            label: "Yaw & Pitch & Roll",
-            value: `this.Yaw & " & " & this.Pitch & " & " & this.Roll`,
+        };
+        // __PUBLISH_EXTRACT_END__
+
+        // Ensure that derived `bis.ViewDefinition` instances along with their properties are also selected.
+        const content = await Presentation.presentation.getContent({
+          imodel,
+          rulesetOrId: ruleset,
+          keys: new KeySet(),
+          descriptor: {},
+        });
+        expect(content!.descriptor.fields).to.containSubset([
+          { label: "Category Selector" },
+          { label: "Code" },
+          { label: "Description" },
+          { label: "Display Style" },
+          { label: "Extents" },
+          { label: "Eye Point" },
+          { label: "Focus Distance" },
+          { label: "Is Camera On" },
+          { label: "Is Private" },
+          { label: "Lens Angle" },
+          { label: "Model" },
+          { label: "Model Selector" },
+          { label: "Origin" },
+          { label: "Pitch" },
+          { label: "Roll" },
+          { label: "User Label" },
+          { label: "Yaw" },
+        ]).and.to.have.lengthOf(17);
+
+        expect(content!.contentSet.length).to.eq(4);
+      });
+
+      it("uses `instanceFilter` attribute", async () => {
+        // __PUBLISH_EXTRACT_START__ ContentInstancesOfSpecificClasses.instanceFilter.Ruleset
+        // The specification returns content of all filtered `bis.SpatialViewDefinition` instances
+        // whose `Pitch` property value is higher or equal to 0.
+
+        const ruleset: Ruleset = {
+          id: "example",
+          rules: [{
+            ruleType: RuleTypes.Content,
+            specifications: [{
+              specType: ContentSpecificationTypes.ContentInstancesOfSpecificClasses,
+              classes: { schemaName: "BisCore", classNames: ["SpatialViewDefinition"] },
+              instanceFilter: "this.Pitch >= 0",
+            }],
           }],
-        }],
-      };
-      // __PUBLISH_EXTRACT_END__
-
-      // Ensure content contains the calculated property and correct value
-      const content = await Presentation.presentation.getContent({
-        imodel,
-        rulesetOrId: ruleset,
-        keys: new KeySet([{ className: "BisCore:Element", id: "0x61" }]),
-        descriptor: {},
+        };
+        // __PUBLISH_EXTRACT_END__
+
+        // Ensure that only `bis.SpatialViewDefinition` instances that have Pitch >= 0 are selected.
+        const content = await Presentation.presentation.getContent({
+          imodel,
+          rulesetOrId: ruleset,
+          keys: new KeySet(),
+          descriptor: {},
+        });
+
+        expect(content!.contentSet.length).to.eq(2);
+        const calculatedField = tryGetFieldByLabel(content!.descriptor.fields, "Pitch");
+        content!.contentSet.forEach(function (contentSetInstance) {
+          expect(Number(contentSetInstance.displayValues[calculatedField!.name])).to.be.not.below(0);
+        });
```

Comment:
I suggest using raw values for this kind of checking:
```suggestion
        content!.contentSet.forEach((record) => {
          expect(record.values[field.name]).to.be.not.below(0);
        });
```